core/matching/request_template_store_test.go

```diff
@@ -386,5 +386,220 @@ func TestRequestTemplateResponsePairCanBeConvertedToARequestResponsePairView_Whi
 	Expect(pairView.Request.Scheme).To(BeNil())
 	Expect(pairView.Request.Query).To(BeNil())
 
-	Expect(pairView.Response.Body).To(Equal("Yo"))
+	Expect(pairView.Response.Body).To(Equal("template matched"))
 }
+
+func TestTemplatesCanUseGlobsOnDestinationAndBeMatched(t *testing.T) {
+	RegisterTestingT(t)
+
+	requestTemplateResponsePair := RequestTemplateResponsePair{
+		RequestTemplate: RequestTemplate{
+			Destination: StringToPointer("*.com"),
+		},
+		Response: models.ResponseDetails{
+			Body: "template matched",
+		},
+	}
+
+	store := RequestTemplateStore{requestTemplateResponsePair}
+
+	request := models.RequestDetails{
+		Method:      "GET",
+		Destination: "testhost.com",
+		Path:        "/api/1",
+	}
+
+	response, err :=store.GetResponse(request, false)
+	Expect(err).To(BeNil())
+
+	Expect(response.Body).To(Equal("template matched"))
+}
+
+func TestTemplatesCanUseGlobsOnPathAndBeMatched(t *testing.T) {
+	RegisterTestingT(t)
+
+	requestTemplateResponsePair := RequestTemplateResponsePair{
+		RequestTemplate: RequestTemplate{
+			Path: StringToPointer("/api/*"),
+		},
+		Response: models.ResponseDetails{
+			Body: "template matched",
+		},
+	}
+
+	store := RequestTemplateStore{requestTemplateResponsePair}
+
+	request := models.RequestDetails{
+		Method:      "GET",
+		Destination: "testhost.com",
+		Path:        "/api/1",
+	}
+
+	response, err :=store.GetResponse(request, false)
+	Expect(err).To(BeNil())
+
+	Expect(response.Body).To(Equal("template matched"))
+}
+
+func TestTemplatesCanUseGlobsOnMethodAndBeMatched(t *testing.T) {
+	RegisterTestingT(t)
+
+	requestTemplateResponsePair := RequestTemplateResponsePair{
+		RequestTemplate: RequestTemplate{
+			Method: StringToPointer("*T"),
+		},
+		Response: models.ResponseDetails{
+			Body: "template matched",
+		},
+	}
+
+	store := RequestTemplateStore{requestTemplateResponsePair}
+
+	request := models.RequestDetails{
+		Method:      "GET",
+		Destination: "testhost.com",
+		Path:        "/api/1",
+	}
+
+	response, err :=store.GetResponse(request, false)
+	Expect(err).To(BeNil())
+
+	Expect(response.Body).To(Equal("template matched"))
+}
+
+func TestTemplatesCanUseGlobsOnSchemeAndBeMatched(t *testing.T) {
+	RegisterTestingT(t)
+
+	requestTemplateResponsePair := RequestTemplateResponsePair{
+		RequestTemplate: RequestTemplate{
+			Scheme: StringToPointer("H*"),
```

Comment:
From this test it appears that the globbing is case-insensitive, which would be find for the method etc, but could it cause any edge cases? (and does this violate the principle of least surprise e.g. with grep I have to turn on case-insensitivity)?